### DISCUSSION TITLE
How do I enable timing summaries for CTSM ports using NUOPC?

### DISCUSSION THREAD

**Discussion by glemieux** [ESCOMP/CTSM#1918] [Q&A] [answered]
This came up in the ctsm software meeting today when @rgknox noted that his port of CTSM wasn't generating any timing files.  I confirmed I was missing the same thing on `lobata`.  The error can be seen in the post run print statement as:

```
Starting job script case.run
Starting job script case.st_archive
No timing file found in run directory
```

**Comment by glemieux** [comment, marked as answer]
NUOPC uses the [ESMF built in profiler](http://earthsystemmodeling.org/docs/release/latest/ESMF_refdoc/node6.html#SECTION060130000000000000000), which is off by default (and doesn't appear to be able to turn ON during ESMF library installation). As such, most machines turn this on in the [config_machines.xml file](https://github.com/ESMCI/ccs_config_cesm/blob/38c6ce4f4bae83c0e4eb6ec68310de4b37d79ec5/machines/config_machines.xml#L831-L833) (I neglected to do this with `lobata` when I added it to the configuration file).  

To enable the timing summaries you can add the following to your specific machine configuration:
```
    <environment_variables comp_interface="nuopc">
      <env name="ESMF_RUNTIME_PROFILE">ON</env>
      <env name="ESMF_RUNTIME_PROFILE_OUTPUT">SUMMARY</env>
    </environment_variables>
```

**Reply by billsacks** [reply]
Ah, nice find @glemieux - thanks!
